client/state/sites/domains/assembler.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * External dependencies
+ */
+import moment from 'moment';
```

Comment:
While I agree, the date object was not decorated in selector in this PR and so `state/sites/domains` is not really usable atm. I'm trying to add the domain notice [here](https://github.com/Automattic/wp-calypso/pull/7860#issuecomment-245557962) using Redux state and the selectors introduced here (nice work btw!) but the domain object I get from a selector is missing a few props such as `registrationMoment` and `name`.

I'm going to create a new PR for this.